src/ja/piccoma/src/eu/kanade/tachiyomi/extension/ja/piccoma/Piccoma.kt

```diff
@@ -0,0 +1,245 @@
+    }
+
+    override fun latestUpdatesParse(response: Response): MangasPage {
+        val document = response.asJsoup()
+        val mangas = document.select("li a:has(div.PCOM-prdList_info)").map { element ->
+            SManga.create().apply {
+                url = element.attr("href")
+                title = element.selectFirst(".PCOM-prdList_title span")!!.text()
+                element.selectFirst("img")?.absUrl("src")?.let { thumbnail_url = it }
+            }
+        }
+        val hasNextPage = document.select("#js_nextPage").isNotEmpty()
+        return MangasPage(mangas, hasNextPage)
+    }
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        if (query.isNotEmpty()) {
+            val url = "$baseUrl/web/search/result_ajax/list".toHttpUrl().newBuilder()
+                .addQueryParameter("word", query)
+                .addQueryParameter("page", page.toString())
+                .addQueryParameter("tab_type", "T")
+                .build()
+            return GET(url, super.headers.newBuilder().add("x-requested-with", "XMLHttpRequest").build())
+        }
+        val rankingPath = filters.firstInstance<RankingFilter>().toUriPart()
+        return GET("$baseUrl/web/ranking/$rankingPath", headers)
+    }
+
+    override fun searchMangaParse(response: Response): MangasPage {
+        if (response.request.url.toString().contains("/ranking/")) {
+            return popularMangaParse(response)
+        }
+
+        val result = response.parseAs<SearchResponseDto>()
+        val mangas = result.data.products.map {
+            SManga.create().apply {
+                url = "/web/product/${it.id}"
+                title = it.title
+                thumbnail_url = "https:${it.img}"
+            }
+        }
+        val currentPage = response.request.url.queryParameter("page")!!.toInt()
+        val hasNextPage = currentPage < result.data.totalPage
+        return MangasPage(mangas, hasNextPage)
+    }
+
+    override fun mangaDetailsParse(response: Response): SManga {
+        val document = response.asJsoup()
+        return SManga.create().apply {
+            title = document.selectFirst("h1.PCM-productTitle")!!.text()
+            author = document.select("ul.PCM-productAuthor li a").joinToString { it.text() }
+            genre = document.select("ul.PCM-productGenre li a, .PCM-productDesc_tagList li a").joinToString { it.text() }
+            description = document.selectFirst("div.PCM-productDesc > p")?.text()
+            document.selectFirst("img.PCM-productThum_img")?.absUrl("src")?.let { thumbnail_url = it }
+            status = when {
+                document.selectFirst("ul.PCM-productStatus")?.text()
+                    ?.contains("連載中") == true -> SManga.ONGOING
+
+                document.selectFirst("ul.PCM-productStatus")?.text()
+                    ?.contains("完結") == true -> SManga.COMPLETED
+
+                else -> SManga.UNKNOWN
```

Comment:
Maybe extract `document.selectFirst("ul.PCM-productStatus")?.text()` to above the `when` block?